### PR TITLE
Encrypted media test google migration: syntax

### PR DESCRIPTION
**Pull request by squapp**
this adds following migrated test cases:

encrypted-media/clearkey-mp4-syntax-mediakeys.html
encrypted-media/clearkey-mp4-syntax-mediakeysession.html
encrypted-media/clearkey-mp4-syntax-mediakeysystemaccess.html

encrypted-media/drm-mp4-syntax-mediakeys.html
encrypted-media/drm-mp4-syntax-mediakeysession.html
encrypted-media/drm-mp4-syntax-mediakeysystemaccess.html

Originally the "syntax" test was only one file. For clarity the tests are now split into 3 files for Clear Key and DRM, respectively.

### CODE REVIEW

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+        {
+            exception: 'TypeError',
+            func: function (mk1) {
+                return mk1.createSession('temporary').load();
+            }
+        },
+        // 'temporary' sessions are never allowed, so always return
+        // 'InvalidAccessError'.
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: InvalidAccessError should be TypeError for non persistent sessions (and for persistent sessions with invalid sessionIds).

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+            func: function (mk3) {
+                return mk3.createSession('temporary').load('');
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+            func: function (mk4) {
+                return mk4.createSession('temporary').load(1);
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+            func: function (mk5) {
+                return mk5.createSession('temporary').load('!@#$%^&*()');
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
+            func: function () {
+                return navigator.requestMediaKeySystemAccess(new Uint8Array(0), [{}]);
+            }
+        },
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: NotSupportedError here as well.

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        // Non-ASCII names.
+        {
+            exception: 'NotSupportedError',
+            func: function () {
+                return navigator.requestMediaKeySystemAccess(keysystem + '\u263A', [{}]);
+            }
+        },
+        // Empty sequence of MediaKeySystemConfiguration.
+        {
+            exception: 'InvalidAccessError',
```
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: I believe this is NotSupportedError too.

**Comment by jdsmith3000** [comment]
With error type comments aside, the rest of these tests LGTM.  Please hold merging until after review by AlexeyKor.

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 8]
```diff
@@ -0,0 +1,489 @@
...
+            return mediaKeySession.generateRequest(type, initData);
+        } catch (e) {
+            assert_true(testCase.isNotSupportedAllowed);
+        }
+    }
+    function generateRequestForVariousSessions(){
+        return new Promise(function(resolve, reject){
+            isInitDataTypeSupported(keysystem, initDataType)
+                .then(function (isTypeSupported) {
+                    assert_true(isTypeSupported, "initDataType not supported");
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: Update description: initDataType should be supported.

File: encrypted-media/scripts/syntax-mediakeys.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+                return mk.setServerCertificate(new Uint8Array(0));
+            }
+        }
+    ];
+
+
+    function setServerCertificateTestExceptions() {
+        return new Promise(function(resolve, reject) {
+            isInitDataTypeSupported(keysystem, initDataType)
+                    .then(function (isTypeSupported) {
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: This applies to all "then"/"catch" in these files.

File: encrypted-media/scripts/syntax-mediakeys.js [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+                    })
+                    .then(function () {
+                        resolve();
+                    })
+                    .catch(function (error) {
+                        reject(error);
+                    });
+        })
+    }
+    promise_test(function(){
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: Could you separate all functions with empty lines?

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 11]
```diff
@@ -0,0 +1,489 @@
...
+                        var mediaKeySession = mediaKeys.createSession();
+                        return mediaKeySession.generateRequest(initDataType, initData).then(function (result) {
+                            return test_exception(testCase, mediaKeySession);
+                        });
+                    });
+                    assert_not_equals(mp4SessionPromises.length, 0);
+                    return Promise.all(mp4SessionPromises);
+                })
+                .then(function () {
+                    resolve()
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 12]
```diff
@@ -0,0 +1,489 @@
...
+                })
+                .then(function (mediaKeys) {
+                    var mp4SessionPromises = kCreateSessionTestCases.map(function (testCase) {
+                        return test_generateRequest(testCase, mediaKeys, initDataType, initData);
+                    });
+                    assert_not_equals(mp4SessionPromises.length, 0);
+                    return Promise.all(mp4SessionPromises);
+                })
+                .then(function () {
+                   resolve()
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 13]
```diff
@@ -0,0 +1,489 @@
...
+                })
+                .catch(function (error) {
+                   reject(error)
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 14]
```diff
@@ -0,0 +1,489 @@
...
+                })
+                .then(function (mediaKeys) {
+                    var sessionPromises = kLoadExceptionsTestCases.map(function (testCase) {
+                        return test_exception(testCase, mediaKeys);
+                    });
+                    assert_not_equals(sessionPromises.length, 0);
+                    return Promise.all(sessionPromises);
+                })
+                .then(function () {
+                    resolve()
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 15]
```diff
@@ -0,0 +1,489 @@
...
+                })
+                .then(function (mediaKeys) {
+                    var mp4SessionPromises = kTypeSpecificGenerateRequestExceptionsTestCases.map(function (testCase) {
+                        return test_exception(testCase, mediaKeys, initDataType, initData);
+                    });
+                    assert_not_equals(mp4SessionPromises.length, 0);
+                    return Promise.all(mp4SessionPromises);
+                })
+                .then(function (result) {
+                    resolve()
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+                    return navigator.requestMediaKeySystemAccess(keysystem, [configuration]);
+                }).then(function (access) {
+                    assert_not_equals(access, null);
+                    assert_equals(typeof access, 'object');
+                    assert_equals(access.keySystem, keysystem);
+                    assert_equals(typeof access.getConfiguration, 'function');
+                    assert_equals(typeof access.createMediaKeys, 'function');
+                    resolve();
+                }).catch(function(error){
+                    reject(error)
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+                })
+        })
+    }
+    promise_test(function(){
+        return requestMediaKeySystemAccessTestAttributes();
+    }, testname + ' test MediaKeySystemAccess attribute syntax.')
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeys.js [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+                assert_equals(typeof mediaKeys.setServerCertificate, 'function');
+                resolve();
+            }).catch(function (error) {
+                reject(error);
+            });
+        })
+    }
+    promise_test(function() {
+        return createMediaKeysAttributeTest();
+    }, testname + ' test MediaKeys attribute syntax')
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: nit: add ; 
This applies to all promise_tests in these files.

**Review by AlexeyKor — APPROVED** [review]
Looks good to me.

**Comment by jdsmith3000** [comment]
LGTM.  I want to merge this today so that all pending new tests are checked in.  Please comment if there is pending work coming.

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 19]
```diff
@@ -0,0 +1,446 @@
...
+                }).then(function (mediaKeys) {
+                    var mp4SessionPromises = kCreateSessionTestCases.map(function (testCase) {
+                        return test_generateRequest(testCase, mediaKeys, initDataType, initData);
+                    });
+                    assert_not_equals(mp4SessionPromises.length, 0);
+                    return Promise.all(mp4SessionPromises);
+                }).then(function () {
+                   resolve();
+                }).catch(function (error) {
+                   reject(error)
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js [hunk 20]
```diff
@@ -0,0 +1,446 @@
...
+            isInitDataTypeSupported(keysystem, initDataType).then(function (isTypeSupported) {
+                    assert_true(isTypeSupported, "initDataType not supported");
+                    return navigator.requestMediaKeySystemAccess(keysystem, [configuration]);
+                }).then(function (access) {
+                    return access.createMediaKeys();
+                }).then(function (mediaKeys) {
+                    kCreateSessionTestCases.map(function (testCase) {
+                        test_createSession(testCase, mediaKeys);
+                    });
+                    resolve()
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+                    return navigator.requestMediaKeySystemAccess(keysystem, [configuration]);
+                }).then(function (access) {
+                    assert_not_equals(access, null);
+                    assert_equals(typeof access, 'object');
+                    assert_equals(access.keySystem, keysystem);
+                    assert_equals(typeof access.getConfiguration, 'function');
+                    assert_equals(typeof access.createMediaKeys, 'function');
+                    resolve();
+                }).catch(function(error){
+                    reject(error)
```
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

**Review by AlexeyKor — COMMENTED** [review]
Could you fix the last 3 nits?

**Comment by jdsmith3000** [comment]
I see there are missing ";" on three lines.  I'm not sure how to pick that up before merging, since the files are in a migration-3 repository at Fraunhofer Fokus.  I could do the merge and then do a separate commit with the fixes.  Okay?

**Comment by mwatson2** [comment]
You can commit the changes to Fraunhofer's branch:

git remote fokus <url for Fraunhofer fork>
git fetch fokus migration-3
git checkout fokus/migration-3

make the changes and commit them

git push fokus HEAD:migration-3

...Mark

**Comment by jdsmith3000** [comment]
I opened this issue:  https://github.com/w3c/web-platform-tests/issues/3912

**Comment by jdsmith3000** [comment]
@mwatson2 I see your guidance.  I'll work that and close the tracking issue.

**Comment by jdsmith3000** [comment]
This is odd.  The last commit triggered lint errors in a file not touched in the commit and not even in the encrypted-media directory:

$ ./lint
PRESENT-TESTHARNESSCSS: web-animations/interfaces/Animation/effect.html Explicit link to testharness.css present
There was 1 error (PRESENT-TESTHARNESSCSS: 1)

**Comment by mwatson2** [comment]
No, that was not your fault. @jgraham merged a bunch of stuff from some
other repo and introduced this problem into master. So the master branch is
failing lint now.

It was a few hours ago, so disappointing it has not been fixed yet.

...Mark

**Comment by jdsmith3000** [comment]
Will this clear when the root problem is fixed, or must we do something to re-check our PR?

**Comment by jdsmith3000** [comment]
I checked in web-animations and there is currently no effect.html file.

**Comment by mwatson2** [comment]
On Thu, Oct 6, 2016 at 1:25 PM, jdsmith3000 <notifications@github.com>
wrote:

> I checked in web-animations and there is currently no effect.html file.
>

​In master there is:
https://github.com/w3c/web-platform-tests/blob/master/web-animations/interfaces/Animation/effect.html
​

**Comment by jdsmith3000** [comment]
It doesn't make sense to me that an unrelated pull request is blocked on a check that fails on a file already in w3c/master...

**Comment by mwatson2** [comment]
On Thu, Oct 6, 2016 at 1:38 PM, jdsmith3000 <notifications@github.com>
wrote:

> It doesn't make sense to me that an unrelated pull request is blocked on a
> check that fails on a file already in w3c/master...
>
​The CI tools run on what the repo would look like if your PR was merged.
So, because the problem is in master, all PR CI runs of lint will fail.

Obviously this can't happen if the only way things are merged into master
is from PRs, since Git won't let you merge if the CI run fails.

​@jgraham must have merged his content some other way. This also happened a
few weeks back.

...Mark

**Pull request merged**